ietf-wdm-tunnel.tree

```diff
@@ -230,37 +246,56 @@ module: ietf-wdm-tunnel
     +--ro path-in-transceiver
     |  +--ro transponder-id?   uint32
     |  +--ro transceivers* [otsi-carrier-id]
-    |     +--ro otsi-carrier-id     uint16
-    |     +--ro transceiver-id?     uint32
-    |     +--ro rx-channel-power?   power-dbm-or-null
-    |     +--ro rx-total-power?     power-dbm-or-null
+    |     +--ro otsi-carrier-id           uint16
+    |     +--ro transceiver-id?           uint32
+    |     +--ro otsi-carrier-frequency?   union
+    |     +--ro rx-channel-power?         power-dbm-or-null
+    |     +--ro rx-total-power?           power-dbm-or-null
+    |     +--ro estimated-gsnr?           snr
+    |     +--ro estimated-eol-gsnr?       snr
+    |     +--ro estimated-lowest-gsnr?    snr
     +--ro path-out-transceiver
        +--ro transponder-id?   uint32
        +--ro transceivers* [otsi-carrier-id]
-          +--ro otsi-carrier-id     uint16
-          +--ro transceiver-id?     uint32
-          +--ro rx-channel-power?   power-dbm-or-null
-          +--ro rx-total-power?     power-dbm-or-null
-  augment /te:te/te:lsps/te:lsp/te:lsp-actual-route-information:
-    +--ro wdm-path-state
-       +--ro gsnr?   l0-types:snr
+          +--ro otsi-carrier-id           uint16
+          +--ro transceiver-id?           uint32
+          +--ro otsi-carrier-frequency?   union
+          +--ro rx-channel-power?         power-dbm-or-null
+          +--ro rx-total-power?           power-dbm-or-null
+          +--ro estimated-gsnr?           snr
+          +--ro estimated-eol-gsnr?       snr
+          +--ro estimated-lowest-gsnr?    snr
   augment /te:te/te:lsps/te:lsp/te:lsp-actual-route-information
             /te:lsp-actual-route-information/te:type
             /te:numbered-node-hop/te:numbered-node-hop:
     +--ro path-in-transceiver
     |  +--ro transponder-id?   uint32
     |  +--ro transceivers* [otsi-carrier-id]
-    |     +--ro otsi-carrier-id     uint16
-    |     +--ro transceiver-id?     uint32
-    |     +--ro rx-channel-power?   power-dbm-or-null
-    |     +--ro rx-total-power?     power-dbm-or-null
+    |     +--ro otsi-carrier-id           uint16
+    |     +--ro transceiver-id?           uint32
+    |     +--ro otsi-carrier-frequency?   union
+    |     +--ro rx-channel-power?         power-dbm-or-null
+    |     +--ro rx-total-power?           power-dbm-or-null
+    |     +--ro pre-fec-ber?              decimal64
+    |     +--ro q-factor?                 int32
```

Comment:
- type for pre-fec-ber and q-factor should be decimal64-or-null
- Add separate timestamp (in minutes?) for pre-fec-ber and q-factor

Reply:
Hello I confirm that pre-fec-ber and q-factor contain redondant information. One is sufficient (I prefer pre-fec-ber :)).
if the two are reported for the same time stamp they should be consistent. if they are not, this can be a symptom of something hapenning on the physical layer that needs further investigation. So my preference would be to keep both, and not force any consistency by the model (the values are transferred as is from the SBI interface, and the usage of the data is left for the applications that consume this model), as we would maybe miss some issues.

Reply:
in l0-types only decimal-2 and decimal-5 are defined. For pre-fec-ber it is usually 18 decimals. We need to define decimal-18/decimal-18-or-null in l0-types as well.